packages/components/src/tabs/src/components/osds-tabs/core/controller.ts

```diff
@@ -1,52 +1,80 @@
 import type { OsdsTabs } from '../osds-tabs';
 
-import { OsdsTabBarItem } from '../../osds-tab-bar-item/osds-tab-bar-item';
-import { OsdsTabsPanel } from '../../osds-tab-panel/osds-tab-panel';
-
 /**
  * common controller logic for text component used by the different implementations.
  * it contains all the glue between framework implementation and the third party service.
  */
 class OdsTabsController {
   protected component: OsdsTabs;
 
+  private _tabItems: (Array<OsdsTabBarItem & HTMLElement>) = [];
+
+  public get tabItems(): (Array<OsdsTabBarItem & HTMLElement>) {
+    return this._tabItems;
+  }
+
+  public set tabItems(value: (Array<OsdsTabBarItem & HTMLElement>)) {
+    this._tabItems = value;
+  }
+
+  private _tabPanels: (Array<OsdsTabsPanel & HTMLElement>) = [];
+
+  public get tabPanels(): (Array<OsdsTabsPanel & HTMLElement>) {
+    return this._tabPanels;
+  }
+
+  public set tabPanels(value: (Array<OsdsTabsPanel & HTMLElement>)) {
+    this._tabPanels = value;
+  }
+
   constructor(component: OsdsTabs) {
     this.component = component;
   }
 
-  beforeInit() {
+  beforeInit(): void {
     this.changeActivePanel(this.component.panel);
     this.component.onContrastedPropChange(this.component.contrasted);
   }
 
-  changeActivePanel(panel: string) {
-    const items = this.component.getTabItems();
+  changeActivePanel(panel: string): void {
+    const items = this.tabItems;
     // if not item found, select the first one
-    if (!items.find((item) => item.panel === panel) && items.length) {
-      panel = items[0].panel;
+    if (!items.find((item) => item.getAttribute('panel') === panel) && items.length) {
+      panel = items[0].getAttribute('panel')!;
     }
-    items.forEach((item) => item.active = item.panel === panel);
-    this.component.getTabPanels().forEach((tabPanel) => tabPanel.active = tabPanel.name === panel);
+    items.forEach((item) => item.active = item.getAttribute('panel') === panel);
+    this.tabPanels.forEach((tabPanel) => tabPanel.active = tabPanel.name === panel);
     if (this.component.panel !== panel) {
       this.component.panel = panel;
       this.component.emitChanged();
     }
   }
 
-  propagateContrastedToItems(contrasted: boolean) {
-    this.component.getTabItems().forEach((item) => item.contrasted = contrasted);
-  }
+  handleArrowKey(event: KeyboardEvent): void {
+    const currentSelectedTabIndex = this.tabItems.findIndex((tab) => tab.hasAttribute('active'));
+    if(currentSelectedTabIndex === 0 && event.code === 'ArrowLeft' ) {
+      return;
+    }
+    if(currentSelectedTabIndex === this.tabItems.length - 1 && event.code === 'ArrowRight' ) {
+      return;
+    }
 
-  getTabItems(elementTag: string) {
-    return Array.from(this.component.el.querySelectorAll<OsdsTabBarItem & HTMLElement>(elementTag));
+    if(event.code === 'ArrowLeft') {
+      const previousPanel = this.tabItems[currentSelectedTabIndex - 1].getAttribute('panel');
```

Comment:
this will fail if `currentSelectedTabIndex` is undefined or less than 1, add more check
<img width="663" alt="Capture d’écran 2023-12-26 à 16 33 26" src="https://github.com/ovh/design-system/assets/1390934/e192a06d-b6f2-4a34-b126-e1fec58482c9">

Reply:
Fixed

Reply:
```suggestion
handleArrowKey(event: KeyboardEvent): void {
    const currentSelectedTabIndex = this.tabItems.findIndex((tab) => tab.hasAttribute('active'));
    
    if (currentSelectedTabIndex === -1) {
      return;
    }

    if (currentSelectedTabIndex === 0 && event.code === 'ArrowLeft') {
      return;
    }

    if (currentSelectedTabIndex === this.tabItems.length - 1 && event.code === 'ArrowRight') {
      return;
    }

    if (currentSelectedTabIndex > 0 && this.tabItems.length > currentSelectedTabIndex - 1) {
      if (event.code === 'ArrowLeft') {
        const previousPanel = this.tabItems[currentSelectedTabIndex - 1].getAttribute('panel');
        this.changeActivePanel(previousPanel!);
      } else {
        const nextPanel = this.tabItems[currentSelectedTabIndex + 1].getAttribute('panel');
        this.changeActivePanel(nextPanel!);
      }
    }
  }
```